File: src/main/java/eotree/EOByte.java
package eotree;

/**
 * EBNF representation:
 * <code>
 * /[\dA-F][\dA-F]/
 * </code>
 */
public class EOByte extends EONode {
    Byte b;

    public EOByte(Byte b) {
        this.b = b;
    }

    @Override
    public String generateEO(int indent) {
        return String.format("%02x", b);
    }
}

File: src/main/kotlin/org/polystat/j2eo/eotree/data/EOByte.kt
package org.polystat.j2eo.eotree.data

import org.polystat.j2eo.eotree.EONode
import java.util.Locale

/**
 * EBNF representation:
 * `
 * /[\dA-F][\dA-F]/
` *
 */
class EOByte(var b: Byte) : EONode() {
    override fun generateEO(indent: Int): String {
        return String.format("%02x", b).uppercase(Locale.getDefault())
    }
}
